src/components/AddPaymentCard/PaymentCardChangeCurrencyForm.tsx

```diff
@@ -0,0 +1,160 @@
+import InputWrapper from '@components/Form/InputWrapper';
+import type {FormInputErrors, FormOnyxValues} from '@components/Form/types';
+import MenuItemWithTopDescription from '@components/MenuItemWithTopDescription';
+import SelectionList from '@components/SelectionList';
+import RadioListItem from '@components/SelectionList/RadioListItem';
+import Text from '@components/Text';
+import TextInput from '@components/TextInput';
+import TextLink from '@components/TextLink';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as ValidationUtils from '@libs/ValidationUtils';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import INPUT_IDS from '@src/types/form/ChangeBillingCurrencyForm';
+import PaymentCardCurrencyModal from './PaymentCardCurrencyModal';
+
+type PaymentCardFormProps = {
+    initialCurrency?: ValueOf<typeof CONST.CURRENCY>;
+    isSecurityCodeRequired?: boolean;
+    changeBillingCurrency: (currency?: ValueOf<typeof CONST.CURRENCY>, values?: FormOnyxValues<typeof ONYXKEYS.FORMS.CHANGE_BILLING_CURRENCY_FORM>) => void;
+};
+
+const REQUIRED_FIELDS = [INPUT_IDS.SECURITY_CODE];
+
+function HeaderCurrencyInfo({isSectionList}: {isSectionList?: boolean}) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    return (
+        <View style={[isSectionList && styles.mh5]}>
+            <Text style={[styles.mt3, isSectionList && styles.mb5]}>
+                {`${translate('billingCurrency.note')}`}{' '}
+                <TextLink
+                    style={styles.link}
+                    href={CONST.PRICING}
+                >{`${translate('billingCurrency.noteLink')}`}</TextLink>{' '}
+                {`${translate('billingCurrency.notDetails')}`}
+            </Text>
+        </View>
+    );
+}
+
+function PaymentCardChangeCurrencyForm({changeBillingCurrency, isSecurityCodeRequired, initialCurrency}: PaymentCardFormProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+
+    const [isCurrencyModalVisible, setIsCurrencyModalVisible] = useState(false);
+    const [currency, setCurrency] = useState<ValueOf<typeof CONST.CURRENCY>>(initialCurrency ?? CONST.CURRENCY.USD);
+
+    const validate = (values: FormOnyxValues<typeof ONYXKEYS.FORMS.CHANGE_BILLING_CURRENCY_FORM>): FormInputErrors<typeof ONYXKEYS.FORMS.CHANGE_BILLING_CURRENCY_FORM> => {
+        const errors = ValidationUtils.getFieldRequiredErrors(values, REQUIRED_FIELDS);
+
+        if (values.securityCode && !ValidationUtils.isValidSecurityCode(values.securityCode)) {
+            errors.securityCode = translate('addPaymentCardPage.error.securityCode');
+        }
+
+        return errors;
+    };
+
+    const {sections} = useMemo(
+        () => ({
+            sections: [
+                {
+                    data: (Object.keys(CONST.CURRENCY) as Array<ValueOf<typeof CONST.CURRENCY>>).map((currencyItem) => ({
```

Comment:
I don't like that we're type casting here. Type casting kind of defeats the point of using TypeScript. Can we update this to not use a an `as` cast?

Reply:
I also don't like that - but generally we do not have `enums` which will correctly be used as a type here. We trying to fit const as enum - that's why such casting necessary here. Without it - it will not work because `currency` will have `string` type - which will not be equal `ValueOf<typeof CONST.CURRENCY>`

if we want to make currency as a string we need to remove everywhere `ValueOf<typeof CONST.CURRENCY>` and just use `string`
Let me know what to do here @blimpich

Reply:
I see, lets keep it as is then. Its not ideal but its our best option right now without doing some refactoring that I don't think is worthwhile. Thank you for the explanation!